src/main/plugins/websearch-plugin/websearch-plugin.ts

```diff
@@ -36,20 +38,46 @@ export class WebSearchPlugin implements ExecutionPlugin {
                         return -1;
                     }
                     return 0;
-                })
-                .map((webSearchEngine): SearchResultItem => {
-                    return {
-                        description: this.buildDescription(webSearchEngine, userInput),
-                        executionArgument: this.buildExecutionArgument(webSearchEngine, userInput),
+                });
+            const promises = webSearchEnginesArray.map((webSearchEngine) => {
+                return axios({ method: "GET", url: webSearchEngine.suggestionUrl.replace("{{query}}", this.getSearchTerm(webSearchEngine, userInput)) });
+            });
+
+            axios.all(promises).then(axios.spread((...responses) => {
+                let responseCounter = 0;
+                responses.forEach((res) => {
+                    const results = res.data[1];
+                    searchResults.push({
+                        description: this.buildDescription(webSearchEnginesArray[responseCounter], userInput),
+                        executionArgument: this.buildExecutionArgument(webSearchEnginesArray[responseCounter], userInput),
                         hideMainWindowAfterExecution: true,
-                        icon: isValidIcon(webSearchEngine.icon) ? webSearchEngine.icon : defaultWebSearchIcon,
-                        name: webSearchEngine.name,
+                        icon: isValidIcon(webSearchEnginesArray[responseCounter].icon) ? webSearchEnginesArray[responseCounter].icon : defaultWebSearchIcon,
+                        name: userInput.replace(webSearchEnginesArray[responseCounter].prefix, ""),
                         originPluginType: this.pluginType,
                         searchable: [],
-                    };
-                });
+                    });
+                    results.some((suggestion: string) => {
+                        searchResults.push(
+                            {
 
-            resolve(searchResults);
+                                description: this.buildDescription(webSearchEnginesArray[responseCounter], suggestion),
+                                executionArgument: this.buildExecutionArgument(webSearchEnginesArray[responseCounter], suggestion),
+                                hideMainWindowAfterExecution: true,
+                                icon: isValidIcon(webSearchEnginesArray[responseCounter].icon) ? webSearchEnginesArray[responseCounter].icon : defaultWebSearchIcon,
+                                name: suggestion,
+                                originPluginType: this.pluginType,
+                                searchable: [],
+                            });
+                        if (results.length > 8) {
+                            return suggestion === results[7];
+                        }
```

Comment:
Can you explain what is happening here?

Reply:
When I was writing this code, I was considering that there might be multiple search engines with the same prefix so I limited each search engine to 8 results only (I think this is enough) so user can see other results from other engines. 

The `if` check is a bad code. I should remove it.